app/src/main/java/com/aaronjwood/portauthority/activity/MainActivity.java

```diff
@@ -146,16 +146,16 @@ public void onClick(View v) {
 
                 hosts.clear();
                 hostsAdapter.notifyDataSetChanged();
-
+                int numberOfHosts = wifi.getNumberOfHostsInWifiSubnet();
                 scanProgressDialog = new ProgressDialog(MainActivity.this, R.style.DialogTheme);
                 scanProgressDialog.setCancelable(false);
                 scanProgressDialog.setTitle("Scanning For Hosts");
                 scanProgressDialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
                 scanProgressDialog.setProgress(0);
-                scanProgressDialog.setMax(255);
+                scanProgressDialog.setMax(numberOfHosts);
                 scanProgressDialog.show();
 
-                Discovery.scanHosts(wifi.getInternalWifiIpAddress(), MainActivity.this);
+                Discovery.scanHosts(wifi.getInternalWifiIpAddress(), Integer.toString(numberOfHosts), MainActivity.this);
```

Comment:
Arguments to `doInBackground(String... params)` in `ScanHostsAsyncTask` only allow a single type.  Hence the casting back and forth. I was going to change it to an int for both types (the IP/hosts) but I think if we pass more options in, it might be better to create an object than pass around data structures.